build.gradle

```diff
@@ -133,3 +142,11 @@ tasks.war.doFirst {
     properties.putAll(hoistMetaData)
     infoFile.withOutputStream {properties.store(it, null)}
 }
+
+// Ensure that all variables defined in .env.template are set in local .env
+tasks.bootRun.doFirst {
+    def missingEnvVars = env.allVariablesOrNull().findAll {it.value == null}.collect {it.key}
+    if (missingEnvVars) {
+        throw new GradleException("Environment variables listed in .env.template not set in local .env file as required: ${missingEnvVars}")
+    }
+}
```

Comment:
Realized that this will only catch the case of vars that are not listed in the local `.env` file at all (including the case where that file has not been created, in which case all vars in the template will be listed).

Does not catch the case where the user has an empty assignment, e.g. `APP_TOOLBOX_DB_PASSWORD=`.  That var will be there, just with a value of empty string. 

In general there's no such thing as an env var having a value of `null` - that's a special feature of the `allVariablesOrNull()` method, where it returns null for vars in the template that are missing from `.env`.

We could tighten this up to also throw on empty strings, but I dunno if we want that - a user might have a blank local MySQL password.  We would also need to limit to `APP_TOOLBOX` prefix only, as you might have some other empty env var on your machine.

I think it's fine as it is - ensures you create a local file, and if we ever add a new var to the template will catch that too.  Just noting as it confused me this morning.